e2e/machineset_migration_test.go

```diff
@@ -213,13 +214,676 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
+		var mapiMSAuthCAPIName = "ms-authoritativeapi-capi"
+		var mapiMSAuthMAPICAPI = "ms-mapi-machine-capi"
+
+		var awsMachineTemplate *capav1.AWSMachineTemplate
+		var capiMachineSet *capiv1beta1.MachineSet
+		var mapiMachineSet *mapiv1beta1.MachineSet
+		var firstMAPIMachine *mapiv1beta1.Machine
+		var secondMAPIMachine *mapiv1beta1.Machine
+		var err error
+
+		Context("with spec.authoritativeAPI: MAPI", Ordered, func() {
+			BeforeAll(func() {
+				mapiMachineSet, err = createMAPIMachineSetWithAuthoritativeAPI(ctx, cl, 1, mapiMSAuthMAPIName, mapiv1beta1.MachineAuthorityMachineAPI, mapiv1beta1.MachineAuthorityMachineAPI)
+				Expect(err).ToNot(HaveOccurred(), "MAPI MachineSet creation should succeed")
+
+				capiMachineSet, awsMachineTemplate = verifyMAPIMachineSetHasCAPIMirror(cl, mapiMSAuthMAPIName)
+
+				mapiMachines, err := mapiframework.GetMachinesFromMachineSet(ctx, cl, mapiMachineSet)
+				Expect(err).ToNot(HaveOccurred(), "failed to get MAPI Machines from MachineSet")
+				capiMachines, err := capiframework.GetMachinesFromMachineSet(cl, capiMachineSet)
+				Expect(err).ToNot(HaveOccurred(), "failed to get CAPI Machines from MachineSet")
+				Expect(capiMachines[0].Name).To(Equal(mapiMachines[0].Name))
+				firstMAPIMachine = mapiMachines[0]
+
+				DeferCleanup(func() {
+					By("Cleaning up Context 'with spec.authoritativeAPI: MAPI' resources")
+					cleanupTestResources(
+						ctx,
+						cl,
+						[]*capiv1beta1.MachineSet{capiMachineSet},
+						[]*capav1.AWSMachineTemplate{awsMachineTemplate},
+						[]*mapiv1beta1.MachineSet{mapiMachineSet},
+					)
+				})
+			})
+
+			It("should scale MAPI MachineSet to 2 replicas succeed", func() {
+				By("Scaling up MAPI MachineSet to 2 replicas")
+				Expect(mapiframework.ScaleMachineSet(mapiMachineSet.GetName(), 2)).To(Succeed(), "should be able to scale up MAPI MachineSet")
+				mapiframework.WaitForMachineSet(ctx, cl, mapiMSAuthMAPIName)
+				Eventually(komega.Object(mapiMachineSet)).Should(HaveField("Spec.Replicas", HaveValue(Equal(int32(2)))))
+
+				By("Verifying a new MAPI Machine is created and Paused condition is False")
+				Eventually(func() (*mapiv1beta1.Machine, error) {
+					secondMAPIMachine, err = mapiframework.GetLatestMachineFromMachineSet(ctx, cl, mapiMachineSet)
+					if err != nil {
+						return nil, err
+					}
+					return secondMAPIMachine, nil
+				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(SatisfyAll(
+					HaveField("Status.Phase", HaveValue(Equal(string(mapiv1beta1.PhaseRunning)))),
+					HaveField("Status.Conditions", ContainElement(SatisfyAll(
+						HaveField("Type", Equal(MAPIPausedCondition)),
+						HaveField("Status", Equal(corev1.ConditionFalse)),
+					))),
+					HaveField("Status.AuthoritativeAPI", Equal(mapiv1beta1.MachineAuthorityMachineAPI)),
+				))
+
+				By("Verifying there is a mirrored CAPI Machine Paused condition is True")
+				Eventually(func() (*capiv1beta1.Machine, error) {
```

Comment:
We reuse this same eventually block 5 times in these tests, this could also be pulled out into a helper func and reused :)

Reply:
We already do this for MAPI, it seems!